docs/Custom-Protos.md

```diff
@@ -0,0 +1,167 @@
+# Creating custom protobuffer messages
+
+Unity and Python communicate by sending protobuffer messages to and from each other. You can create custom protobuffer messages if you want to exchange structured data beyond what is included by default. 
+
+Whenever you change the fields of a custom message, you must run `protobuf-definitions/make.bat` to create C# and Python files corresponding to the new message. Follow the directions in that file for guidance. After running it, reinstall the Python package by running `pip install ml-agents` and make sure your Unity project is using the newly-generated version of `UnitySDK`.
```

Comment:
`make.bat` is missing two key pieces of information to get this to work:

- You have to use the right version of Grpc.Tools (1.14.1) when installing with nuget. Otherwise the .cs script will be generated but it will not compile in Unity. 
- You have to have `protobuf` installed (e.g. via brew or apt) before running the script. This one is pretty obvious. 
-

Reply:
That is true. Since that's just as true with or without this PR though, should adding that information be a separate PR?

Reply:
I think we should add it to this PR since normally there is no reason to use make.bat - but now make.bat is a user-facing interface. @vincentpierre, what do you think?